stable/application-mesh/templates/redirects.yaml

```diff
@@ -0,0 +1,79 @@
+{{- range $_, $redirect := .Values.redirects }}
+{{- $full_url := cat $redirect.source "" $redirect.source_uri -}}
+---
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: {{ include "application-helpers.name" $ }}-{{ include "application-helpers.uniq-id" $full_url }}-redirects
+spec:
+  hosts:
+    - {{ $redirect.source }}
+  gateways:
+    - {{ $.Release.Namespace }}/{{ $redirect.source | replace "." "-" }}-redirects
+  http:
+    - match:
+        - uri:
+            prefix: {{ $redirect.source_uri | default "/" }}
+      redirect:
+        authority: {{ $redirect.destination }}
+        {{- with $redirect.destination_uri }}
+        uri: {{ . }}
+        {{- end }}
+        redirectCode: {{ $redirect.redirectCode | default "301" }}
+{{ end }}
+
+{{/* We need Certificate and Gateway for each unique domain
+in the redirects list */}}
+
+{{- $uniqueSources := dict }}
```

Comment:
There is `uniq` function available in helm: https://helm.sh/docs/chart_template_guide/function_list/#uniq-mustuniq. Might be easier to get uniq list of domains